src/sync.js

```diff
@@ -30,136 +30,201 @@ function decrypt(key, encrypted) {
   });
 }
 
+// An "id schema" used to validate Kinto IDs and generate new ones.
+const notesIdSchema = {
+  // FIXME: Maybe this should generate IDs?
+  generate() {
+    throw new Error("cannot generate IDs");
+  },
 
-function loadFromKinto(client) {
+  validate(id) {
+    // FIXME: verify that at least this matches Kinto server ID format
+    return true;
+  },
+};
+
+class ServerKeyNewerError extends Error {
+  constructor() {
+    super("key used to encrypt the record appears to be newer than our key");
+  }
+}
+
+class ServerKeyOlderError extends Error {
+  constructor() {
+    super("key used to encrypt the record appears to be older than our key");
+  }
+}
+
+class JWETransformer {
+  constructor(key) {
+    this.key = key;
+  }
+
+  async encode(record) {
```

Comment:
I think ESLint is choking here and failing to parse the entire file... Should this be `async function encode(record) {`?

Reply:
Ah, is it maybe the case that you guys stick to ES2015 on this project? Or which standard do you support? Per http://kangax.github.io/compat-table/es2016plus/#test-async_functions_async_methods,_classes, this syntax is supported in FF 52 ESR and greater, which is why it's used in the storage.sync implementation, which is why I copied and pasted it here :blush: